<launch>
    <arg name="fps" default="30" />
    <arg name="device" default="cuda:0" />
    <arg name="use_bag" default="true" />
    <arg name="bag" default="$(find detect)/outputs/bags/robot_20.bag" />
    <arg name="enable_depth_filter" default="true" />
    <arg name="enable_candidate_filter" default="true" />
    <arg name="robot_name" default="myrobot" />
    <arg name="image_topic" default="/$(arg robot_name)/body_camera/color/image_raw" />
    <arg name="depth_topic" default="/$(arg robot_name)/body_camera/aligned_depth_to_color/image_raw" />
    <arg name="image_info_topic" default="/$(arg robot_name)/body_camera/aligned_depth_to_color/camera_info" />
    <arg name="depth_info_topic" default="/$(arg robot_name)/body_camera/aligned_depth_to_color/camera_info" />
    <arg name="seg_topic" default="/$(arg robot_name)/body_camera/color/image_raw/segmented_img" />
    <arg name="objects_topic" default="/$(arg robot_name)/body_camera/color/image_raw/detected_objects" />
    <arg name="candidates_topic" default="/$(arg robot_name)/body_camera/color/image_raw/candidates_img" />


    <!-- for realsense-viewer bags -->
    <!-- <remap from="/device_0/sensor_1/Color_0/image/data" to="$(arg image_topic)" />
    <remap from="/device_0/sensor_1/Color_0/info/camera_info" to="$(arg image_info_topic)" />
    <remap from="/device_0/sensor_0/Depth_0/image/data" to="$(arg depth_topic)" />
    <remap from="/device_0/sensor_0/Depth_0/info/camera_info" to="$(arg depth_info_topic)" /> -->
    <!-- <remap from="/device_0/sensor_0/Depth_0/depth/data" to="/body_camera/depth/image_raw" /> -->
    <!-- <remap from="/device_0/sensor_0/Depth_0/info/camera_info" to="/body_camera/depth/camera_info" /> -->

    <param name="use_sim_time" value="true" />

    <param name="fps" value="$(arg fps)" />
    <param name="device" value="$(arg device)" />
    <param name="enable_depth_filter" value="$(arg enable_depth_filter)" />
    <param name="enable_candidate_filter" value="$(arg enable_candidate_filter)" />
    <param name="robot_name" value="$(arg robot_name)" />
    <param name="image_topic" value="$(arg image_topic)" />
    <param name="depth_topic" value="$(arg depth_topic)" />
    <param name="image_info_topic" value="$(arg image_info_topic)" />
    <param name="depth_info_topic" value="$(arg depth_info_topic)" />
    <param name="seg_topic" value="$(arg seg_topic)" />
    <param name="objects_topic" value="$(arg objects_topic)" />
    <param name="candidates_topic" value="$(arg candidates_topic)" />

    <!-- <node name="raw_view" pkg="rqt_image_view" type="rqt_image_view" args="/body_camera/color/image_raw" /> -->
    <node name="depth_view" pkg="rqt_image_view" type="rqt_image_view" args="$(arg depth_topic)" />
    <node name="seg_view" pkg="rqt_image_view" type="rqt_image_view" args="$(arg seg_topic)" />
    <node name="cnds_view" pkg="rqt_image_view" type="rqt_image_view" args="$(arg candidates_topic)" />

    <group if="$(arg use_bag)">
        <node name="rosbag_play" pkg="rosbag" type="play" args="$(arg bag) -r $(arg fps) -l --clock" />
        <node name="grasp_detection_test_client" pkg="detect" type="test_grasp_detection.py" output="screen" />
    </group>

    <!-- action server -->
    <node name="tf_transform_server" pkg="detect" type="tf_transform_server.py" />
    <node name="visualize_server" pkg="detect" type="visualize_server.py" />
    <node name="instance_segmentation_server" pkg="detect" type="instance_segmentation_server.py" />
    <node name="grasp_detection_server" pkg="detect" type="grasp_detection_server.py" output="screen" />
    <group if="$(arg enable_depth_filter)">
        <node name="compute_depth_threshold_server" pkg="detect" type="compute_depth_threshold_server.py" output="screen" />
    </group>
</launch>